nsq/reader.py

```diff
@@ -702,14 +709,31 @@ def process_message(self, message):
 
     def giving_up(self, message):
         """
-        Called when a message has been received where ``msg.attempts > max_tries``
+        Called when a message has been received where ``msg.attempts > max_tries``.
+        
+        Failed messages will be archived to ``$data_path/$channel_$topic/%Y%m%d-%H%M%S-%f_$sequence.failed.msg``
 
-        This is useful to subclass and override to perform a task (such as writing to disk, etc.)
+        This is useful to subclass and override to perform a custom task (such as writing to disk, etc.)
 
         :param message: the :class:`nsq.Message` received
         """
-        logger.warning('[%s] giving up on message %s after %d tries (max:%d) %r',
-                       self.name, message.id, message.attempts, self.max_tries, message.body)
+
+        self.failed_count += 1
+        
+        path = os.path.join(self.data_path or "", self.topic + '_' + self.channel)
```

Comment:
We should probably use a separator that isn't a valid character for a topic/channel.  Perhaps the channel should just be a real subdir?